charts/argo-cd/templates/_helpers.tpl

```diff
@@ -183,7 +183,7 @@ Argo Configuration Preset Values (Influenced by Values configuration)
 {{- define "argo-cd.config.cm.presets" -}}
 {{- $presets := dict -}}
 {{- $_ := set $presets "url" (printf "https://%s" .Values.global.domain) -}}
-{{- if index .Values.configs.cm "statusbadge.enabled" | eq true -}}
+{{- if eq (toString (index .Values.configs.cm "statusbadge.enabled")) "true" -}}
```

Comment:
Existing examples of supporting "true" .
* https://github.com/yu-croco/argo-helm/blob/master/charts/argo-cd/templates/argocd-server/clusterrole.yaml#L31